kvdb/etcd/commit_queue.go

```diff
@@ -36,19 +43,20 @@ func NewCommitQueue(ctx context.Context) *commitQueue {
 		ctx:       ctx,
 		readerMap: make(map[string]int),
 		writerMap: make(map[string]int),
-		queue:     make(chan func(), commitQueueSize),
+		queue:     list.New(),
+		shutdown:  make(chan struct{}),
 	}
+	q.queueCond = sync.NewCond(&q.queueMx)
 
 	// Start the queue consumer loop.
-	q.wg.Add(1)
 	go q.mainLoop()
 
 	return q
 }
 
 // Wait waits for the queue to stop (after the queue context has been canceled).
 func (c *commitQueue) Wait() {
-	c.wg.Wait()
+	c.signalUntilShutdown()
```

Comment:
Isn't this more of a `Stop()` method? There's no explicit `Start()` method. But `Wait()` somehow suggests this might be called multiple times during its lifecycle when in fact the queue won't be working after this was called.

Reply:
nit: could also inline the content of `signalUntilShutdown` here since it's only ever called from here.

Reply:
done